File: anagram/ana.go
package main

import (
	"bufio"
	"fmt"
	"os"
	"sort"
	"strings"
)

func sortString(str string) string {
	var runeSlice []rune

	for _, value := range str {
		runeSlice = append(runeSlice, value)
	}

	sort.SliceStable(runeSlice, func(i, j int) bool {
		return runeSlice[i] < runeSlice[j]
	})

	return string(runeSlice)
}

func contains(strSlice []string, str string) bool {
	for _, v := range strSlice {
		if v == str {
			return true
		}
	}
	return false
}

func main() {
	theDict := make(map[string][]string)

	file, err := os.Open("wordlist.txt")

	if err != nil {
		fmt.Println(err)
	}

	defer file.Close()

	words := bufio.NewScanner(file)

	for words.Scan() {
		word := strings.Trim(strings.Replace(words.Text(), "'", "", -1), "\n")

		if len(word) < 2 {
			continue
		}

		sortedWord := sortString(word)

		if _, keyPresent := theDict[sortedWord]; keyPresent {
			if !contains(theDict[sortedWord], word) {
				theDict[sortedWord] = append(theDict[sortedWord], word)
			}
		} else {
			theDict[sortedWord] = []string{}
		}
	}

	counter := 0
	for k, v := range theDict {
		if len(v) >= 10 {
			counter += 1
			fmt.Printf("[%d] %s has %d words: %v\n", counter, k, len(v), v)
		}
	}
}

File: anagram/ana.rs
use std::{
    collections::HashMap,
    fs::File,
    io::{BufRead, BufReader},
};

fn main() {
    let mut the_dict: HashMap<String, Vec<String>> = HashMap::new();

    let file = File::open("wordlist.txt").unwrap();
    let mut words = BufReader::new(file).lines();

    while let Some(Ok(word)) = words.next() {
        let word = word.trim().replace("'", "");

        if word.len() < 2 {
            continue;
        }

        let mut word_vec: Vec<char> = word.chars().collect();
        word_vec.sort();
        let sorted_word: String = word_vec.into_iter().collect();

        if the_dict.contains_key(&sorted_word) {
            if !the_dict[&sorted_word].contains(&word) {
                the_dict.get_mut(&sorted_word).unwrap().push(word);
            }
        } else {
            the_dict.insert(sorted_word, vec![]);
        }
    }

    let mut counter = 0;
    for (k, v) in the_dict {
        if v.len() >= 10 {
            counter += 1;
            println!("[{}] {} has {} words: {:?}", counter, k, v.len(), v);
        }
    }
}
